examples/bgp/main.go

```diff
@@ -23,7 +29,18 @@ func main() {
 		log.Fatal(err)
 	}
 
+	go startMetricsEndpoint(b)
+
 	startServer(b, v)
 
 	select {}
 }
+
+func startMetricsEndpoint(server server.BGPServer) {
+	prometheus.MustRegister(prom_bgp.NewCollector(server))
+	prometheus.MustRegister(prom_vrf.NewCollector())
+
+	http.Handle("/metrics", promhttp.Handler())
+	logrus.Info("Metrics are available :8080/metrics")
+	log.Fatal(http.ListenAndServe(":8080", nil))
```

Comment:
Do we want to shut down the complete process if a auxiliary function like metrics fails? We should log the error. But the operator should decide how to act in case of an failed metrics endpoint.

Reply:
It's the example. so I would not mind. In fact in the config based full featured routing daemon (which will come later) we should not terminate. In this case we should even implement a retry mechanism / auto heal. We can not afford to restart the daemon to get our metrics back. For the example file I will change it to log.Error without any further handling.